code/datums/wires/_wires.dm

```diff
@@ -279,6 +281,28 @@
 /datum/wires/proc/always_reveal_wire(color)
 	return FALSE
 
+/**
+ * Attempts to study a photo for blueprints.
+ */
+/datum/wires/proc/try_study_photo(mob/user)
+	if(randomize)
+		return
+
+	for(var/obj/item/photo/photo in user.held_items)
+		if(!photo.picture?.has_blueprints)
+			continue
+
+		var/study_length = 1 SECONDS * floor(min(photo.picture.psize_x, photo.picture.psize_y) / 32)
+		if(study_length >= 4 SECONDS)
+			to_chat(user, span_notice("<i>You squint [photo]... Hey, there's blueprints in the frame! Really wish the photo was zoomed in, though. \
+				It's rather difficult to make out the wires.</i>"))
+		else
+			to_chat(user, span_notice("<i>You glance at [photo], looking for wires in the pictured blueprints.</i>"))
+
+		if(do_after(user, study_length, holder, hidden = TRUE))
+			LAZYSET(studied_photos, REF(user), REF(photo))
```

Comment:
Wouldn't it more suitable to register a ref to the mind datum and then check that?